src/core/components/model.jsx

```diff
@@ -66,16 +66,20 @@ export default class Model extends ImmutablePureComponent {
     /*
      * If we have an unresolved ref, get the schema and name from the ref.
      * If the ref is external, we can't resolve it, so we just display the ref location.
-     * This is for situations where the ref was not resolved by Swagger Client
-     * because we reached the traversal depth limit.
+     * This is for situations where: 
+     *  - the ref was not resolved by Swagger Client because we reached the traversal depth limit
+     *  - we had a circular ref inside the allOf keyword
      */
     if ($ref) {
-      name = this.getModelName($ref)
-      const refSchema = this.getRefSchema(name)
+      const refName = this.getModelName($ref)
+      const refSchema = this.getRefSchema(refName)
       if (Map.isMap(refSchema)) {
-        schema = refSchema.set("$$ref", $ref)
-        $$ref = $ref
-      } else {
+        schema = refSchema.mergeDeep(schema) 
```

Comment:
Merging `schema` into `refSchema` so that the same properties from `refSchema` won't overwrite those in `schema`, for example in case of `title`.

Reply:
Would ignoring other keywords make sense if they come from other merged schemas (as in, from the `allOf` merge)?

I did it this way as the other way around we would still have these as titles:
<img width="579" alt="Screenshot 2024-04-09 at 16 05 51" src="https://github.com/swagger-api/swagger-ui/assets/44094416/f63a6776-a9bf-4113-8c40-c2cd2df03f66">

Because we would merge the title from the unresolved reference in `allOf`, here:
```
FirstOneOf:
      title: FirstOneOf
      type: object
      allOf:
        -
          $ref: '#/components/schemas/OneOfParent'
        -
          properties:
            numberProp:
              type: number
              example: '1'
          type: object
```

And this is with the current order:
<img width="553" alt="Screenshot 2024-04-09 at 16 06 10" src="https://github.com/swagger-api/swagger-ui/assets/44094416/101509ef-be77-49df-afa7-a488eab594c5">